storage/replica.go

```diff
@@ -1011,6 +1015,7 @@ func (r *Replica) applyRaftCommand(ctx context.Context, index uint64, originRepl
 		// time it's required.
 		if _, ok := ba.GetArg(roachpb.TruncateLog); ok {
 			r.setCachedTruncatedState(nil)
+			r.raftEntryCache.clearAll()
```

Comment:
Why do you clear the entire cache here on `TruncateLog` when you also clear part of it inside the `Replica.TruncateLog` method?